hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/SimpleAvroKeyGenerator.java

```diff
@@ -47,6 +50,9 @@ public SimpleAvroKeyGenerator(TypedProperties props) {
 
   @Override
   public String getRecordKey(GenericRecord record) {
+    if (recordKeyFields.isEmpty()) {
```

Comment:
lets make sure all Key gen code paths are taken care of, including the Spark Row based key gen

Reply:
ack. double checking.